### PR TITLE
Fix/improve renderer validations

### PR DESCRIPTION
**Pull request by Vin-RM**
Improve existing validations and add remaining.
Improve checkbox logic.

### CODE REVIEW

File: packages/value_renderer/lib/src/widgets/inputs/text_input.dart [hunk 1]
```diff
@@ -76,11 +78,11 @@ class TextInputState extends State<TextInput> {
   }
 
   String? validateInput(input) {
-    if (input.isEmpty) {
+    if (input.isEmpty && widget.mustBeAccepted == true) {
       return FlutterI18n.translate(context, 'errors.value_renderer.emptyField');
-    } else if (!validateEquality(input)) {
+    } else if (!input.isEmpty && !validateEquality(input)) {
```
Comment (nicole-eb) [inline packages/value_renderer/lib/src/widgets/inputs/text_input.dart:83]: ``
```suggestion
    } else if (input.isNotEmpty && !validateEquality(input)) {
```

File: packages/value_renderer/lib/src/widgets/inputs/text_input.dart [hunk 2]
```diff
@@ -76,11 +78,11 @@ class TextInputState extends State<TextInput> {
...
       return FlutterI18n.translate(context, 'errors.value_renderer.invalidInput');
-    } else if (!validateFormat(input)) {
+    } else if (!input.isEmpty && !validateFormat(input)) {
```
Comment (nicole-eb) [inline packages/value_renderer/lib/src/widgets/inputs/text_input.dart:85]:
```suggestion
    } else if (input.isNotEmpty && !validateFormat(input)) {
```

File: packages/value_renderer/lib/src/widgets/inputs/number_input.dart [hunk 3]
```diff
@@ -88,18 +90,18 @@ class NumberInputState extends State<NumberInput> {
...
-    if (input.isEmpty) {
+    if (input.isEmpty && widget.mustBeAccepted == true) {
       return FlutterI18n.translate(context, 'errors.value_renderer.emptyField');
     } else {
       final numInput = num.parse(input);
 
       if (!validateMaxValue(numInput)) {
         return FlutterI18n.translate(context, 'errors.value_renderer.maxValue', translationParams: {'value': widget.max!.toString()});
-      } else if (!validateMinValue(numInput)) {
+      } else if (!input.isEmpty && !validateMinValue(numInput)) {
```
Comment (nicole-eb) [inline packages/value_renderer/lib/src/widgets/inputs/number_input.dart:100]:
```suggestion
      } else if (input.isNotEmpty && !validateMinValue(numInput)) {
```

File: packages/value_renderer/lib/src/widgets/inputs/number_input.dart [hunk 4]
```diff
@@ -88,18 +90,18 @@ class NumberInputState extends State<NumberInput> {
...
         return FlutterI18n.translate(context, 'errors.value_renderer.minValue', translationParams: {'value': widget.min!.toString()});
-      } else if (!validateFormat(input)) {
+      } else if (!input.isEmpty && !validateFormat(input)) {
```
Comment (nicole-eb) [inline packages/value_renderer/lib/src/widgets/inputs/number_input.dart:102]:
```suggestion
      } else if (input.isNotEmpty && !validateFormat(input)) {
```

File: packages/value_renderer/lib/src/widgets/inputs/number_input.dart [hunk 5]
```diff
@@ -88,18 +90,18 @@ class NumberInputState extends State<NumberInput> {
...
         return FlutterI18n.translate(context, 'errors.value_renderer.invalidFormat');
-      } else if (!validateEquality(numInput)) {
+      } else if (!input.isEmpty && !validateEquality(numInput)) {
```
Comment (nicole-eb) [inline packages/value_renderer/lib/src/widgets/inputs/number_input.dart:104]:
```suggestion
      } else if (input.isNotEmpty && !validateEquality(numInput)) {
```

**Review by nicole-eb — CHANGES_REQUESTED** [review]

**Review by nicole-eb — APPROVED** [review]

File: packages/renderers/lib/src/request/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart [hunk 6]
```diff
@@ -176,4 +180,12 @@ class _DecidableReadAttributeRequestItemRendererState extends State<DecidableRea
...
     }
   }
+
+  void updateCheckbox() {
+    if (widget.controller != null && !widget.item.mustBeAccepted) {
+      setState(() {
+        isChecked = true;
+      });
+    }
+  }
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart:190]: I don't like the naming of this method. when I see the name I don't get why and when it is updated. When I look what the method does I see that it only gets checked - so no update just enabling.

Additionally this method should not be public.

**Review by jkoenig134 — CHANGES_REQUESTED** [review]
Only one small point. the rest LGTM

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**